static/js/topic_list.js

```diff
@@ -159,7 +178,41 @@ export class TopicListWidget {
         return this.my_stream_id;
     }
 
+    update_topic_search_text() {
+        const input = this.parent_elem.find("#filter-topic-input");
+        if (input.length) {
+            this.topic_search_text = input.val();
+        } else {
+            this.topic_search_text = "";
+        }
+    }
+
+    update_topic_search_input() {
+        const input = this.parent_elem.find("#filter-topic-input");
+        if (input.length) {
+            // update search text.
+            input.val(this.topic_search_text);
+            input.trigger("focus");
+
+            // setup event handlers.
+            const rebuild_list = () => this.build();
+            function clear_topic_search(e) {
+                e.stopPropagation();
+                input.val("");
+                input.trigger("blur");
+                rebuild_list();
+                return;
+            }
+            $("#clear_search_topic_button").on("click", clear_topic_search);
+            input.on("input", rebuild_list);
+        }
+    }
+
     remove() {
+        // If topic search was on, we store the
+        // input value lazily just before removing
+        // old dom.
+        this.update_topic_search_text();
```

Comment:
Do we also need code to clear the `this.topic_search_text` when exiting the topic zoom state?

Reply:
Before removing old dom we update `this.topic_search_text` and if no topic filter input was present in the view we clear input. So when going from normal view to zoomed view as in old dom(normal view) no topic filter input was present we clear it. So we are kinda already doing this.